src/components/DetailsModal.tsx

```diff
@@ -72,44 +90,153 @@ function DetailsModal({
+            color={displayValues.pillColor}
+            icon={displayValues.pillIcon}
+          />
+        </span>
+        <span className="font-sans-md padding-bottom-2">
+          Stage: {details.lastService} - {details.lastAction}
+        </span>
+      </>
+    );
   };
 
-  const getContent = () => {
-    const messages = getErrorMessages();
-    if (messages.length > 0) {
-      return (
-        <Alert heading="Failed" type="error">
-          {messages.length} Error(s) were detected
-          {messages.map((issue: string) => (
-            <p
-              key={issue}
-              className="border-1px radius-md border-secondary-light margin-y-2 padding-105">
-              {issue}
-            </p>
-          ))}
-        </Alert>
-      );
-    }
+  const getFailedContent = () => {
+    if (details.status.toLowerCase() != "failed") return null;
 
-    if (submission.status.toLowerCase().includes("complete")) {
-      return (
-        <Alert heading="Complete" type="success">
-          File submission is complete
-        </Alert>
-      );
-    }
+    const issues = details.reports[0]?.issues;
 
-    if (submission.status.toLowerCase().includes("uploading")) {
+    if (!issues.length)
       return (
-        <Alert heading="Processing" type="info">
-          File submission is processing
+        <Alert
+          className="margin-y-1"
+          heading={`Error in stage ${details.lastService} - ${details.lastAction}`}
+          type="error">
+          No specifics were given, please see full attached report below.
         </Alert>
       );
-    }
+
+    return (
+      <Alert
+        className="margin-y-1"
+        heading={`${issues.length} error(s) returned in stage ${details.lastService} - ${details.lastAction}`}
+        type="error">
+        Please review and address these errors.
+        {issues.map((issue: Issue) => (
+          <p
+            key={issue.message}
+            className="border-1px radius-md border-secondary-light margin-y-2 padding-105">
+            {issue.message}
+          </p>
+        ))}
+      </Alert>
+    );
+  };
+
+  const hasFailedReports = () => {
+    const failedReports = details.reports.filter(
+      (r: Report) => r.status == "FAILURE"
+    );
+    return failedReports.length > 0;
+  };
+
+  const getIssues = () => {
+    const issueSummary: IssueSummary[] = details.reports.flatMap(
+      (report: Report) => {
+        const internalIssues: IssueSummary[] = report.issues.map(
+          (issue: Issue) => ({
+            service: report.service,
+            action: report.action,
+            level: issue.level,
+            message: issue.message,
+          })
+        );
+
+        return internalIssues;
+      }
+    );
+
+    return (
+      <Accordion
+        multiselectable
+        items={issueSummary.map((issue: IssueSummary, i: number) => {
+          const item: AccordionItemProps = {
+            id: i.toString(),
+            title: (
+              <div
```

Comment:
Maybe make a component for this, like `RowTitle`

Reply:
Moved the entire summary to separate component.